components/prism-wolfram.js

```diff
@@ -0,0 +1,24 @@
|$HTTPRequest|$IgnoreEOF|$ImageFormattingWidth|$ImageResolution|$ImagingDevice|$ImagingDevices|$ImportFormats|$IncomingMailSettings|$InitialDirectory|$Initialization|$InitializationContexts|$Input|$InputFileName|$InputStreamMethods|$Inspector|$InstallationDate|$InstallationDirectory|$InterfaceEnvironment|$InterpreterTypes|$IterationLimit|$KernelCount|$KernelID|$Language|$LaunchDirectory|$LibraryPath|$LicenseExpirationDate|$LicenseID|$LicenseProcesses|$LicenseServer|$LicenseSubprocesses|$LicenseType|$Line|$Linked|$LinkSupported|$LoadedFiles|$LocalBase|$LocalSymbolBase|$MachineAddresses|$MachineDomain|$MachineDomains|$MachineEpsilon|$MachineID|$MachineName|$MachinePrecision|$MachineType|$MaxExtraPrecision|$MaxLicenseProcesses|$MaxLicenseSubprocesses|$MaxMachineNumber|$MaxNumber|$MaxPiecewiseCases|$MaxPrecision|$MaxRootDegree|$MessageGroups|$MessageList|$MessagePrePrint|$Messages|$MinMachineNumber|$MinNumber|$MinorReleaseNumber|$MinPrecision|$MobilePhone|$ModuleNumber|$NetworkConnected|$NetworkInterfaces|$NetworkLicense|$NewMessage|$NewOptimizations|$NewSymbol|$NotebookInlineStorageLimit|$Notebooks|$NoValue|$NumberMarks|$NumberOfExtraPages|$Off|$OperatingSystem|$Output|$OutputForms|$OutputSizeLimit|$OutputStreamMethods|$Packages|$ParentLink|$ParentProcessID|$PasswordFile|$PatchLevelID|$Path|$PathnameSeparator|$PerformanceGoal|$Permissions|$PermissionsGroupBase|$PersistenceBase|$PersistencePath|$PipeSupported|$PlotTheme|$Post|$Pre|$PreferencesDirectory|$PreInitialization|$PrePrint|$PreRead|$PrintForms|$PrintLiteral|$Printout3DPreviewer|$ProcessID|$ProcessorCount|$ProcessorType|$ProductInformation|$ProgramName|$ProgressReporting|$PublisherID|$RandomGeneratorState|$RandomState|$RecursionLimit|$RegisteredDeviceClasses|$RegisteredUserName|$ReleaseNumber|$RequesterAddress|$RequesterWolframID|$RequesterWolframUUID|$ResourceSystemBase|$ResourceSystemPath|$RootDirectory|$ScheduledTask|$ScriptCommandLine|$ScriptInputString|$SearchLanguage|$SecuredAuthenticationKeyTokens|$ServiceCreditsAvailable|$Services|$SessionID|$SetParentLink|$SharedFunctions|$SharedVariables|$SoundDisplay|$SoundDisplayFunction|$SourceLink|$SSHAuthentication|$SubtitleDecoders|$SubtitleEncoders|$SummaryBoxDataSizeLimit|$SuppressInputFormHeads|$SynchronousEvaluation|$SyntaxHandler|$System|$SystemCharacterEncoding|$SystemCredentialStore|$SystemID|$SystemMemory|$SystemShell|$SystemTimeZone|$SystemWordLength|$TargetSystems|$TemplatePath|$TemporaryDirectory|$TemporaryPrefix|$TestFileName|$TextStyle|$TimedOut|$TimeUnit|$TimeZone|$TimeZoneEntity|$TopDirectory|$TraceOff|$TraceOn|$TracePattern|$TracePostAction|$TracePreAction|$UnitSystem|$Urgent|$UserAddOnsDirectory|$UserAgentLanguages|$UserAgentMachine|$UserAgentName|$UserAgentOperatingSystem|$UserAgentString|$UserAgentVersion|$UserBaseDirectory|$UserBasePacletsDirectory|$UserDocumentsDirectory|$Username|$UserName|$UserURLBase|$Version|$VersionNumber|$VideoDecoders|$VideoEncoders|$VoiceStyles|$WolframDocumentsDirectory|$WolframID|$WolframUUID)\b/,
```

Comment:
@RunDevelopment  These are the the symbols in the System` context, a strength of this language is that it has an extremely large builtin library of commands. I guess we don't need to color them.  

User-defined symbols can contain numbers but must start with letter, but the builtins are 7108 symbols that the system predefines (Names["System`*"] // Length == 7108) 

I'm not sure what the best practice is here...

Reply:
Maybe we could limit it to builtins that handle control flow (e.g. `If`)? After all, we don't have to highlight every builtin function.

> but must start with letter

But a few in the regex start with `$`?

Reply:
Yes adding the $ prefix is their convention for system defined global variables

Reply:
Ah I misspoke, I meant to say valid variable names _can't start with a number_ but other than that you can use arbitrary unicode.

Reply:
> Yes adding the $ prefix is their convention for system defined global variables

Would that be something worth highlighting on its own?

I'm imaging something like `'variable': /\$\w+/`.

Reply:
Mathematica's front end doesn't do that, but it might be nice feature to add here anyway!

Reply:
It would also reduce the file size again. Thanks to you cutting down the list of builtins, we are down to about 2kBs.

With all variables matched by a simple regex, like the one above, we could probably go down to 1kB.